src/elements/content-sidebar/activity-feed/comment/CommentMenu.js

```diff
@@ -0,0 +1,71 @@
+// @flow
+import * as React from 'react';
+import { injectIntl, FormattedMessage } from 'react-intl';
+import getProp from 'lodash/get';
+
+import IconTrash from '../../../../icons/general/IconTrash';
+import IconEllipsis from '../../../../icons/general/IconEllipsis';
+import IconPencil from '../../../../icons/general/IconPencil';
+
+import PlainButton from '../../../../components/plain-button';
+import DropdownMenu from '../../../../components/dropdown-menu';
+import { Menu, MenuItem } from '../../../../components/menu';
+
+import messages from '../../../common/messages';
+import deleteMessages from '../inline-delete/messages';
+import { ACTIVITY_TARGETS } from '../../../common/interactionTargets';
+import { fours, nines } from '../../../../styles/variables';
+import { COMMENT_TYPE_DEFAULT, COMMENT_TYPE_TASK } from '../../../../constants';
+
+type Props = {
+    id: string,
+    isDisabled: boolean,
+    onDeleteClick?: Function,
+    onEditClick?: Function,
+    permissions?: BoxItemPermission,
+    type: typeof COMMENT_TYPE_DEFAULT | typeof COMMENT_TYPE_TASK,
+} & InjectIntlProvidedProps;
+
+const CommentMenu = (props: Props) => {
+    const { intl, isDisabled, onDeleteClick, onEditClick, permissions, type } = props;
+    const canDelete = getProp(permissions, 'can_delete', false);
```

Comment:
It looks like we perform these checks in the parent, as well. Should they be passed in?

Reply:
simplified this a bit to use a default and falsy values since they are special checks. future menu items might need other permissions